ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v1/teaser/clientlib/editor/js/teaser.js

```diff
@@ -0,0 +1,94 @@
+    const actionsMultifieldSelector = '.cmp-teaser__editor-multifield_actions';
+    const actionsMultifieldItemSelector = '.coral3-Multifield-item';
+    const actionsEnabledCheckboxSelector = 'coral-checkbox[name="./actionsEnabled"]';
+
+    $(document).on('dialog-loaded', e => {
+        const $dialog = e.dialog;
+        const $dialogContent = $dialog.find(dialogContentSelector);
+        const dialogContent = $dialogContent.length > 0 ? $dialogContent[0] : undefined;
+
+        // check if the expected dialog was loaded
+        if (dialogContent) {
+            const multiFieldActions = $dialogContent.find(actionsMultifieldSelector)[0];
+
+            // Handle pickers values change for existing actions
+            handlePickersChange(multiFieldActions);
+
+            // Add change handler to multifield actions
+            multiFieldActions.on('change', () => {
+                // whenever actions are being added/removed, reattach picker change handlers
+                handlePickersChange(multiFieldActions);
+            });
+
+            // Fix Core WCM Components Teaser editor bug.
+            // when actions are disabled, only products picker gets disabled ( Core WCM Components Teaser expects only one action )
+            const $actionsEnabledCheckbox = $dialogContent.find(actionsEnabledCheckboxSelector);
+            $actionsEnabledCheckbox.on('change', e => {
+                const actionsEnabled =
+                    $(e.target)
+                        .adaptTo('foundation-field')
+                        .getValue() === 'true';
+                $(categoryFieldSelector).each((ix, catEl) => {
+                    $(catEl)
+                        .adaptTo('foundation-field')
+                        .setDisabled(!actionsEnabled);
+                });
+            });
+        }
+    });
+
+    // used to handle picker value changes and keep only one picker populated
+    const handlePickersChange = multiFieldActions => {
+        $(multiFieldActions)
+            .find(actionsMultifieldItemSelector)
+            .each((ix, action) => {
+                const productElement = $(action).find(productFieldSelector);
+                const categoryElement = $(action).find(categoryFieldSelector);
+                const productField = productElement.adaptTo('foundation-field');
+                const categoryField = categoryElement.adaptTo('foundation-field');
+
+                // remove attached handlers
+                productElement.off('change', handleProductChange);
+                categoryElement.off('change', handleCategoryChange);
+
+                const eventData = { productField, categoryField };
+                // [re]attach change handlers
+                productElement.on('change', eventData, handleProductChange);
+                categoryElement.on('change', eventData, handleCategoryChange);
+            });
+    };
+
+    // sets an empty value on the category field when product field gets updated
```

Comment:
I have to admit I'm surprised there is so much code in this file, I naively thought we would only need (more or less) these two functions to implement the "XOR fields" (one or the other but not both), with only some "onChange" event handlers. Or what do I miss?

Reply:
the fact that the teaser allows for multiple actions makes it a little more complicated.
but maybe I missed something. let me know if there's a simpler way of doing this